server/src/main/java/org/elasticsearch/transport/Transport.java

```diff
@@ -155,35 +155,15 @@ default Object getCacheKey() {
     }
 
     /**
-     * This class represents a response context that encapsulates the actual response handler, the action and the connection it was
-     * executed on.
+     * This class represents a response context that encapsulates the actual response handler, the action. the connection it was
+     * executed on, and the request ID.
      */
-    final class ResponseContext<T extends TransportResponse> {
-
-        private final TransportResponseHandler<T> handler;
-
-        private final Connection connection;
-
-        private final String action;
-
-        ResponseContext(TransportResponseHandler<T> handler, Connection connection, String action) {
-            this.handler = handler;
-            this.connection = connection;
-            this.action = action;
-        }
-
-        public TransportResponseHandler<T> handler() {
-            return handler;
-        }
-
-        public Connection connection() {
-            return this.connection;
-        }
-
-        public String action() {
-            return this.action;
-        }
-    }
+    record ResponseContext<T extends TransportResponse>(
+        TransportResponseHandler<T> handler,
+        Connection connection,
+        String action,
+        Long requestId
```

Comment:
Why a boxed `Long`? I would expect this never to be `null`?

Reply:
Because I saw it as a key in the Map, but indeed it can be `long` here. Did that.